presidio-analyzer/tests/test_analyzer_engine.py

```diff
@@ -490,3 +491,74 @@ def test_demo_text(self):
         assert len([entity for entity in detected_entities if entity == "US_SSN"]) == 1
 
         assert len(results) == 19
+
+    def test_get_recognizers_returns_predefined(self):
+        analyze_engine = AnalyzerEngine(registry=RecognizerRegistry(),
+                                        nlp_engine=loaded_spacy_nlp_engine)
+        request = RecognizersAllRequest(language="en")
+        response = analyze_engine.GetAllRecognizers(request, None)
+        # there are 15 predefined recognizers that detect the 17 entities
+        assert len(response) == 15
+
+    def test_get_recognizers_returns_custom(self):
+        pattern = Pattern("rocket pattern", r'\W*(rocket)\W*', 0.8)
+        pattern_recognizer = PatternRecognizer("ROCKET",
+                                               name="Rocket recognizer",
+                                               patterns=[pattern])
+
+        recognizers_store_api_mock = RecognizerStoreApiMock()
+        recognizers_store_api_mock.add_custom_pattern_recognizer(
+            pattern_recognizer)
+        analyze_engine = AnalyzerEngine(registry=
+            MockRecognizerRegistry(
+                recognizers_store_api_mock),
+                nlp_engine=MockNlpEngine())
+        request = RecognizersAllRequest(language="en")
+        response = analyze_engine.GetAllRecognizers(request, None)
+        # there are 15 predefined recognizers and one custom
+        assert len(response) == 16
+        rocket_recognizer = [x for x in response if x.name == "Rocket recognizer" 
```

Comment:
It would be nice to have meaningful names. It's hard to read and understand.
`x for x `

Reply:
fixed